src/lava/lib/optimization/apps/scheduler/solver.py

```diff
@@ -0,0 +1,289 @@
+from lava.lib.optimization.apps.scheduler.utils.util_funcs import *
+from lava.lib.optimization.problems.problems import QUBO
+from lava.lib.optimization.solvers.generic.solver import (OptimizationSolver,
+                                                          SolverConfig)
+from lava.lib.optimization.utils.generators.mis import MISProblem
+
+
+class Scheduler:
+
+    def __init__(self,
+                 sp: SchedulingProblem,
+                 qubo_weights: Tuple[int, int] = (1, 8),
+                 probe_cost: bool = False,
+                 probe_loihi_exec_time=False,
+                 probe_loihi_energy=False):
+        """Solver for Scheduling Problems.
+
+        Parameters
+        ----------
+        sp (SchedulingProblem) : Scheduling problem object as defined in
+        lava.lib.optimization.apps.scheduler.problems
+        qubo_weights (tuple(int, int)) : The QUBO weight matrix parameters
+        for diagonal and off-diagonal weights. Default is (1, 8).
+        probe_cost (bool) : Toggle whether to probe cost during the solver
+        run. Default is False.
+        """
+        self._problem = sp
+        self._graph = sp.graph
+        self._qubo_hyperparams = {
+            "temperature": int(8),
+            "refract": np.random.randint(64, 127,
+                                         self._graph.number_of_nodes()),
+            "refract_counter": np.random.randint(0, 64,
+                                                 self._graph.number_of_nodes()),
+        }
+        self._qubo_weights = qubo_weights
+        self._probe_cost = probe_cost
+        self._probe_loihi_exec_time = probe_loihi_exec_time
+        self._probe_loihi_energy = probe_loihi_energy
+        self._netx_solution = None
+        self._qubo_problem = None
+        self._qubo_matrix = None
+        self._lava_backend = 'Loihi2' if loihi.host else 'CPU'
+        self._lava_solver_report = None
+        self._lava_solution = None
+
+        sol_criterion = self._problem.sat_cutoff
+        if type(sol_criterion) is float and 0.0 < sol_criterion <= 1.0:
+            self._qubo_target_cost = int(
+                -sol_criterion * self._problem.num_tasks * qubo_weights[0])
+        elif type(sol_criterion) is int and sol_criterion < 0:
+            self._qubo_target_cost = sol_criterion
+
+    @property
+    def problem(self):
+        return self._problem
+
+    @property
+    def graph(self):
+        return self._graph
+
+    @property
+    def qubo_hyperparams(self):
+        return self._qubo_hyperparams
+
+    @qubo_hyperparams.setter
+    def qubo_hyperparams(self, hp_update: Tuple[Dict, bool]):
+        """
+        Set hyperparameters for QUBO solver
+        Parameters
+        ----------
+        hp_update (tuple of dict and bool) :
+        The bool part toggles whether to update the existing hyperparameters
+            or to set new ones from scratch.
+
+        The dict part of the tuple has the following keys:
```

Comment:
For hyperparameters and default values, could we refer to the QUBO solver? I would not hard code it here. We will forget to change this whenever we change the QUBO solver.

Reply:
Done.